embassy-nrf/src/pwm.rs

```diff
@@ -55,18 +55,18 @@ impl<'d, T: Instance> SequencePwm<'d, T> {
     /// mechanisms) on stack allocated buffers which which have been passed to
     /// [`new()`](SequencePwm::new).
     #[allow(unused_unsafe)]
-    pub fn new<'a>(
+    pub fn new(
         _pwm: impl Unborrow<Target = T> + 'd,
         ch0: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch1: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch2: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch3: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         config: SequenceConfig,
-        sequence: &'a [u16],
+        sequence: [u16; N],
```

Comment:
This is even "more UB" than before: you're now giving DMA a pointer to this *local variable*. I guess you meant to make it a field in Pwm, but that's stilll UB because the user can move the Pwm

Reply:
Yeah was totally not working :)